### PR TITLE
Separate docs and reference

### PR DESCRIPTION
**Pull request by paulomarg**
### WHY are these changes introduced?

Closes https://github.com/Shopify/first-party-library-planning/issues/513

Currently, the documentation in this repo is mostly framed as tutorials, which are not the ideal content for a github repo, and it didn't fully cover all of the methods provided by the library.

### WHAT is this pull request doing?

Breaking the documentation up into tutorials and references, which go into more detail on each of the methods we provide. The content is the same, it's mostly just been reformatted with a couple of rewrites to fit the new format.

I kept the tutorial because there was useful content in there, but we should consider making them proper documentation tutorials in shopify.dev.

To view the final state of the updated docs: https://github.com/Shopify/shopify-api-js/tree/separate_docs_and_reference. I also broke the changes up into commits by section to make it easier to dive into a specific area.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,57 +85,26 @@ const app = express();
...
-  - [Detecting scope changes](docs/usage/oauth.md#detecting-scope-changes)
-- [REST Admin API client](docs/usage/rest.md)
-  - [Using REST resources](docs/usage/rest.md#using-rest-resources)
-- [GraphQL Admin API client](docs/usage/graphql.md)
-- [GraphqL Storefront API client](docs/usage/storefront.md)
-- [Webhooks](docs/usage/webhooks.md)
-  - [Register a Webhook](docs/usage/webhooks.md#register-a-webhook)
-  - [Process a Webhook](docs/usage/webhooks.md#process-a-webhook)
-- [Billing](docs/usage/billing.md)
+## Tutorials
```
Comment (teddyhwang) [inline README.md:101]: When I look at other library docs, I think the term Guides is a more familiar and common pattern.

```suggestion
## Guides
```

Comment (teddyhwang) [reply README.md:101]: With this change, I'd also suggest we change the folder name from `usage` to `guides` to maintain consistency

**Review by teddyhwang — COMMENTED** [review]
this is so much better! one minor feedback

File: README.md [hunk 2]
```diff
@@ -85,57 +85,26 @@ const app = express();
 ### Next steps
 
 Once you configure your app, you can use this package to access the Shopify APIs.
-See the specific documentation in the [Features section](#features) for instructions on how to get API access tokens and use them to query the APIs.
+See the [reference documentation](./docs/reference/README.md) for details on all the methods provided by this package.
+
+See the specific documentation in the [Tutorials section](#tutorials) for high-level instructions on how to get API access tokens and use them to query the APIs.
 
 As a general rule, apps will want to interact with the Admin API to fetch / submit data to Shopify.
```
Comment (darius-gai) [inline README.md:92]: This sentence felt a bit awkward. Curious as the reason for this caution?

File: docs/reference/shopifyApi.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+### [utils](./utils/README.md)
+
+Object containing general functions to help build apps.
+
+### rest
+
+Object containing OO representations of the Admin REST API.
+See the [API reference documentation](https://shopify.dev/api/admin-rest) for details.
+
+[Back to reference index](./README.md)
```
Comment (darius-gai) [inline docs/reference/shopifyApi.md:189]: I really like this link back to the reference index. Could we add it to the other sub reference pages?

File: docs/usage/billing.md [hunk 4]
```diff
@@ -45,151 +33,40 @@ This setting is a collection of billing plans. Each billing plan allows the foll
...
 If you're gating access to the entire app, you should check for billing:
 
-1. After OAuth completes, you'll get the session back from `shopify.auth.callback`. You can use the session to ensure billing takes place as part of the authentication flow.
-1. When validating requests from the frontend. Since the check requires API access, you can only run it in requests that work with `shopify.session.getCurrentId`.
+1. After OAuth completes, you'll get the session back from [`shopify.auth.callback`](../reference/auth/callback.md). You can use the session to ensure billing takes place as part of the authentication flow.
+1. When validating requests from the frontend. Since the check requires API access, you can only run it in requests that work with [`shopify.session.getCurrentId`](../reference/session/getCurrentId.md).
 
 **Note**: the merchant may refuse payment when prompted or cancel subscriptions later on, but the app will already be installed at that point. We recommend using [billing webhooks](https://shopify.dev/apps/billing#webhooks-for-billing) to revoke access for merchants when they cancel / decline payment.
```
Comment (darius-gai) [inline docs/usage/billing.md:72]: Similar to the reference comment, really like the link back to guide index here. Could we add to the other guides?

File: docs/usage/rest-resources.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+});
+
+// use sessionId to retrieve session from app's session storage
+// getSessionFromStorage() must be provided by application
+const session = await getSessionFromStorage(sessionId);
+
+const client = new shopify.clients.Rest({
+  domain: session.shop,
+  accessToken: session.accessToken,
+});
```
Comment (mkevinosullivan) [inline docs/usage/rest-resources.md:36]:
```suggestion
const client = new shopify.clients.Rest({session});
```

File: docs/usage/rest-resources.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+Please visit our [REST API reference documentation](https://shopify.dev/api/admin-rest) for detailed instructions on how to call each of the endpoints.
+
+## Mounting REST resources
+
+The library doesn't include all available resources automatically to make loading it more efficient, since there are hundreds of resource classes.
+To use REST resources, you can import the desired version and mount it onto your `shopify` object when creating it:
+
+```ts
+import {shopifyApi} from '@shopify/shopify-api';
+import {restResources} from '@shopify/shopify-api/rest/admin/2022-07';
```
Comment (mkevinosullivan) [inline docs/usage/rest-resources.md:94]:
```suggestion
import {restResources} from '@shopify/shopify-api/rest/admin/2022-10';
```

File: docs/usage/rest-resources.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+const shopify = shopifyApi({
+  ...,
+  apiVersion: ApiVersion.July22,
```
Comment (mkevinosullivan) [inline docs/usage/rest-resources.md:98]:
```suggestion
  apiVersion: ApiVersion.October22,
```

File: README.md (hunk 2 above)
Comment (paulomarg) [reply README.md:92]: This is just to build context and introduce the guides section, and mention the admin API clients.

File: docs/reference/shopifyApi.md [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+
+Creates a new library object that provides all features needed for an app to interact with Shopify APIs.
+Use this function when you [set up your app](../../README.md#getting-started).
+
+## Example
+
+```ts
+import {shopifyApi, ApiVersion, BillingInterval} from '@shopify/shopify-api';
+import {MemorySessionStorage} from '@shopify/shopify-api/session-storage/memory';
+import {restResources} from '@shopify/shopify-api/rest/admin/2022-07';
```
[the remark this answers is not in the document]
Comment (paulomarg) [reply docs/reference/shopifyApi.md:11]: This needs to match the below value, so it'll eventually become outdated.

File: docs/reference/shopifyApi.md [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+
+`string` | :exclamation: **required**
+
+App host name in the format `my-host-name.com`. Do **not** include the scheme or leading or trailing slashes.
+
+### hostScheme
+
+`"https" | "http"` | Defaults to `"https"`
+
+The scheme for your app's public URL. `http` is only allowed if your app is running on `localhost`.
```
[the remark this answers is not in the document]
Comment (paulomarg) [reply docs/reference/shopifyApi.md:72]: It's not really an issue if it is, but we could consider validating.

File: docs/reference/utils/versionCompatible.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+Its main use is when you want to tweak behaviour depending on your current API version, though apps won't typically need this kind of check.
+
+## Example
+
+```ts
+const shopify = shopifyApi({
+  apiVersion: ApiVersion.July22,
+});
+
+if (shopify.utils.versionCompatible(ApiVersion.January22)) {
```
[the remark this answers is not in the document]
Comment (paulomarg) [reply docs/reference/utils/versionCompatible.md:13]: I like that. We could replace and deprecate this method if we feel that's better. It does read better.

**Review by teddyhwang — APPROVED** [review]

**Pull request merged**